go/libraries/doltcore/sqle/dsess/session.go

```diff
@@ -155,6 +155,7 @@ type DatabaseSessionState struct {
 	readReplica          *env.Remote
 	TempTableRoot        *doltdb.RootValue
 	TempTableEditSession *editor.TableEditSession
+	tmpdir               string
```

Comment:
Do we need this extra member? We have access to a `RepoStateReader` through `dbData`

Reply:
yeah that's what i thought originally, but the session state adaptor is the RsR so you get an infinite recursion if you `dbData.RsR.TempTableFilesDir()`